photutils/psf/photometry.py

```diff
@@ -391,6 +404,41 @@ def _define_fit_param_names(self):
                 self._pars_to_set[p0] = p
                 self._pars_to_output[pfit] = p
 
+    def _get_uncertainties(self, star_group_size):
+        """
+        Retrieve uncertainties on fitted parameters from the fitter object.
+
+        Parameters
+        ----------
+        star_group_size : int
+            Number of stars in the given group.
+
+        Returns
+        -------
+        unc_tab : `~astropy.table.Table`
+            Table which contains uncertainties on the fitted parameters.
+            The uncertainties are reported as one standard deviation.
+        """
+
+        unc_tab = Table()
+        for param_name in self.psf_model.param_names:
+            if not self.psf_model.fixed[param_name]:
+                unc_tab.add_column(Column(name=param_name + "_unc",
+                                          data=np.empty(star_group_size)))
+
+        if 'param_cov' in self.fitter.fit_info.keys():
+            if self.fitter.fit_info['param_cov'] is not None:
+                k = 0
+                n_fit_params = len(unc_tab.colnames)
+                for i in range(star_group_size):
+                    unc_tab[i] = np.sqrt(np.diag(
+                                          self.fitter.fit_info['param_cov'])
+                                         )[k: k + n_fit_params]
+                    k = k + n_fit_params
+        else:
+            unc_tab[:] = np.nan
```

Comment:
To do what I suggested above, you could just return `None` here, and then check for `None` in the call to this above in the loop.